app/controllers/reports_controller.rb

```diff
@@ -1,19 +1,22 @@
 # frozen_string_literal: true
 
 class ReportsController < ApplicationController
-  before_action :set_report, only: %i[show edit update destroy]
-
   def index
     @reports = Report.order(:id).page(params[:page])
   end
 
-  def show; end
+  def show
+    @report = Report.find(params[:id])
+  end
 
   def new
     @report = Report.new
   end
 
-  def edit; end
+  def edit
+    @report = current_user.reports.find_by(id: params[:id])
```

Comment:
ここだけ

```suggestion
    @report = current_user.reports.find(params[:id])
```

にしなかった理由って何かありますか？

Reply:
findで検索できないと例外（ActiveRecord::RecordNotFound）になりエラーページが表示されるため、find_by（のnil応答）を使用しました...🙄